### ISSUE TITLE
Git Flow

### ISSUE THREAD

**Issue by animate1978** [animate1978/MB-Lab#189]
**Is your feature request related to a problem? Please describe.**
Apparently the current development model is more "trunk" based than what Git was designed for. As such I myself have made mistakes even recently that caused the dev branch to be deleted and starting over is the only option. This is bad practice and caused a headache for myself. A better model is needed.

**Describe the solution you'd like**
Using git flow development model as described here : 
https://nvie.com/posts/a-successful-git-branching-model/

In as there should be the following branches

- master
- dev
- release_xxx
- features
- hotfixes

![gitflow_concept_01](https://user-images.githubusercontent.com/479115/68529758-7e5d1b80-02c7-11ea-83e1-89704a7a2aed.png)

**Describe alternatives you've considered**
Git was designed for such a thing, any alternatives are not really options.

**Additional context**
As I use Linux for development primarily there is the git-flow package that I want to read up on before I start making new branches.

In general though I think this is an important issue that the project should begin using.

**Comment by econundrum** [comment]
I think with more people becoming interested in helping with development this does need to become more of a priority. The current system of forking the repository isn't really great and of course the current dev branch is broken.

**Comment by animate1978** [comment]
So once development starts, after the holidays, I will make some changes to documentation and wiki about git-flow. Something I think yes we should adopt.

**Comment by econundrum** [comment]
I read through the link, I think it's a good flow. The only downside is that it suggests feature branches should exist only in the developers local repositories. That's probably ok for small features with a single developer but for larger features we may actually want the branches on here so multiple contributors can work together.

**Comment by animate1978** [comment]
I agree, already I kind of tried this approach when starting on the code refactor, I created a new branch, did my changes, deleted the branch. Not sure if this is the way to go though... maybe a dedicated feature branch for the next version.

So in effect it would be : 

- master
- dev
- feature_178

Then once we get 1.7.8 at a feature freeze we make a release branch for any and all bug fixes

Just throwing ideas out there

**Comment by econundrum** [comment]
That sounds like a good idea to me.
